### ISSUE TITLE
promesa dependencies won't get resolved at runtime

### ISSUE THREAD

**Issue by trieloff** [thheller/shadow-cljs#137]
This is a continuation of #136, using code from here: https://github.com/trieloff/gotta-get-you-onto-my-server

`shadow-cljs compile app` works just fine now, but at runtime, I'm getting following error:

```bash
$ node onto-server.js
SHADOW import error /code/gotta-get-you-onto-my-server/target/cljs-runtime/promesa.impl.js

/code/gotta-get-you-onto-my-server/target/cljs-runtime/promesa/impl.cljc:47
   (def ^:const Promise (js/Promise.noConflict)))
                        ^
TypeError: Promise.noConflict is not a function
    at Object.<anonymous> (/code/gotta-get-you-onto-my-server/target/cljs-runtime/promesa/impl.cljc:47:25)
    at SHADOW_ENV.SHADOW_IMPORT (/code/gotta-get-you-onto-my-server/onto-server.js:104:44)
    at /code/gotta-get-you-onto-my-server/onto-server.js:173:1
    at Object.<anonymous> (/code/gotta-get-you-onto-my-server/onto-server.js:186:3)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:390:7)
    at startup (bootstrap_node.js:150:9)
    at bootstrap_node.js:505:3
```

`Promise.noConflict` is imported from Bluebird: http://bluebirdjs.com/docs/api/promise.noconflict.html but it looks like `Promise` itself is `undefined`

**Comment by thheller** [comment]
I suspect that it is using the native `Promise` which has no `noConflict` method.

**Comment by trieloff** [comment]
Shouldn't https://github.com/funcool/promesa/blob/master/src/promesa/impl.cljc#L28 pull in the bluebird `Promise`? Is there a way to verify this in a repl?

**Comment by thheller** [comment]
Yes.

```
(prn [js/Promise js/goog.global.Promise (identical? js/Promise js/goog.global.Promise)])
[#object[Promise] #object[Promise] false]
```

I think I know what causes this.

**Comment by thheller** [comment]
Yeah confirmed. Some time ago I rewrote the node bootstrapping stuff so it doesn't pollute the `global` namespace. This was done so that 2 different CLJS libs running in the same VM would not interfere with each other. See https://github.com/thheller/shadow-build/issues/30.

But in this case we want to override the global. Might need an option to toggle the "global-free" mode.

**Comment by trieloff** [comment]
You are too fast for me. In the time it takes me to set up npm to start the repl you've already confirmed the issue ;)

**Comment by thheller** [comment]
`shadow-cljs node-repl` ;)

**Comment by trieloff** [comment]
yep: https://github.com/trieloff/gotta-get-you-onto-my-server/commit/cfb8d258d5ff31cfe20c9223232ae03147140725

**Comment by thheller** [comment]
So I think I'm going to back out of this `global` protection. It is really causing more trouble than its worth and no-one should be running 2 different CLJS dev builds in the same VM anyways.

**Comment by thheller** [comment]
I need to think about this for a bit. In the meantime you could use `:npm-module`.

```
shadow-cljs compile npm
```

`onto-server.js`
```
var x = require("shadow-cljs/onto.server");
x.main();
```

It is not as convenient since you have to write the `onto-server.js` file but it doesn't have the `global` protection.

**Comment by thheller** [comment]
Fixed in `shadow-cljs@2.0.90`.
